Tests/SwiftDocCTests/Infrastructure/DocumentationContext/DocumentationContextTests.swift

```diff
@@ -4291,13 +4302,100 @@ let expected = """
             }
         }
     }
-    
+
+    func testOverloadGroupHasSimplifiedSubHeading() throws {
+        enableFeatureFlag(\.isExperimentalOverloadedSymbolPresentationEnabled)
+
+        var symbols = [SymbolGraph.Symbol]()
+        // func myFunc(param: Int, other: Int) -> Int
+        symbols.append(makeSymbol(
+            name: "myFunc(param:other:)",
+            identifier: "myFunc-1",
+            kind: .method,
+            subHeading: [
+                .init(kind: .keyword, spelling: "func", preciseIdentifier: nil),
+                .init(kind: .text, spelling: " ", preciseIdentifier: nil),
+                .init(kind: .identifier, spelling: "myFunc", preciseIdentifier: nil),
+                .init(kind: .text, spelling: "(", preciseIdentifier: nil),
+                .init(kind: .externalParameter, spelling: "param", preciseIdentifier: nil),
+                .init(kind: .text, spelling: ": ", preciseIdentifier: nil),
+                .init(kind: .typeIdentifier, spelling: "Int", preciseIdentifier: "Int"),
+                .init(kind: .text, spelling: ", ", preciseIdentifier: nil),
+                .init(kind: .externalParameter, spelling: "other", preciseIdentifier: nil),
+                .init(kind: .text, spelling: ": ", preciseIdentifier: nil),
+                .init(kind: .typeIdentifier, spelling: "Int", preciseIdentifier: "Int"),
+                .init(kind: .text, spelling: ") -> ", preciseIdentifier: nil),
+                .init(kind: .typeIdentifier, spelling: "Int", preciseIdentifier: "Int"),
+            ]
+        ))
+        // func myFunc(param: String, other: String) -> String
+        symbols.append(makeSymbol(
+            name: "myFunc(param:other:)",
+            identifier: "myFunc-2",
+            kind: .method,
+            subHeading: [
+                .init(kind: .keyword, spelling: "func", preciseIdentifier: nil),
+                .init(kind: .text, spelling: " ", preciseIdentifier: nil),
+                .init(kind: .identifier, spelling: "myFunc", preciseIdentifier: nil),
+                .init(kind: .text, spelling: "(", preciseIdentifier: nil),
+                .init(kind: .externalParameter, spelling: "param", preciseIdentifier: nil),
+                .init(kind: .text, spelling: ": ", preciseIdentifier: nil),
+                .init(kind: .typeIdentifier, spelling: "String", preciseIdentifier: "String"),
+                .init(kind: .text, spelling: ", ", preciseIdentifier: nil),
+                .init(kind: .externalParameter, spelling: "other", preciseIdentifier: nil),
+                .init(kind: .text, spelling: ": ", preciseIdentifier: nil),
+                .init(kind: .typeIdentifier, spelling: "String", preciseIdentifier: "String"),
+                .init(kind: .text, spelling: ") -> ", preciseIdentifier: nil),
+                .init(kind: .typeIdentifier, spelling: "String", preciseIdentifier: "String"),
+            ]
+        ))
```

Comment:
Why make `symbols` mutable here instead of directly initializing the array?
```suggestion
        let symbols = [
            // func myFunc(param: Int, other: Int) -> Int
            makeSymbol(
                name: "myFunc(param:other:)",
                identifier: "myFunc-1",
                kind: .method,
                subHeading: [
                    .init(kind: .keyword, spelling: "func", preciseIdentifier: nil),
                    .init(kind: .text, spelling: " ", preciseIdentifier: nil),
                    .init(kind: .identifier, spelling: "myFunc", preciseIdentifier: nil),
                    .init(kind: .text, spelling: "(", preciseIdentifier: nil),
                    .init(kind: .externalParameter, spelling: "param", preciseIdentifier: nil),
                    .init(kind: .text, spelling: ": ", preciseIdentifier: nil),
                    .init(kind: .typeIdentifier, spelling: "Int", preciseIdentifier: "Int"),
                    .init(kind: .text, spelling: ", ", preciseIdentifier: nil),
                    .init(kind: .externalParameter, spelling: "other", preciseIdentifier: nil),
                    .init(kind: .text, spelling: ": ", preciseIdentifier: nil),
                    .init(kind: .typeIdentifier, spelling: "Int", preciseIdentifier: "Int"),
                    .init(kind: .text, spelling: ") -> ", preciseIdentifier: nil),
                    .init(kind: .typeIdentifier, spelling: "Int", preciseIdentifier: "Int"),
                ]
            ),
            // func myFunc(param: String, other: String) -> String
            makeSymbol(
                name: "myFunc(param:other:)",
                identifier: "myFunc-2",
                kind: .method,
                subHeading: [
                    .init(kind: .keyword, spelling: "func", preciseIdentifier: nil),
                    .init(kind: .text, spelling: " ", preciseIdentifier: nil),
                    .init(kind: .identifier, spelling: "myFunc", preciseIdentifier: nil),
                    .init(kind: .text, spelling: "(", preciseIdentifier: nil),
                    .init(kind: .externalParameter, spelling: "param", preciseIdentifier: nil),
                    .init(kind: .text, spelling: ": ", preciseIdentifier: nil),
                    .init(kind: .typeIdentifier, spelling: "String", preciseIdentifier: "String"),
                    .init(kind: .text, spelling: ", ", preciseIdentifier: nil),
                    .init(kind: .externalParameter, spelling: "other", preciseIdentifier: nil),
                    .init(kind: .text, spelling: ": ", preciseIdentifier: nil),
                    .init(kind: .typeIdentifier, spelling: "String", preciseIdentifier: "String"),
                    .init(kind: .text, spelling: ") -> ", preciseIdentifier: nil),
                    .init(kind: .typeIdentifier, spelling: "String", preciseIdentifier: "String"),
                ]
            ),
        ]
```

Reply:
Purely because i wanted to save the level of indentation. 😅 I can change it if this is more desired.

Reply:
For me, it's mostly about what a mutable variable signals. I kept looking later in the code thinking "surely there is another value getting added somewhere later" and not finding any.